### PR TITLE
 feat: multiple artifacts for GitHub and Shellable 

### PR DESCRIPTION
**Pull request by rix0rrr**
Add multiple output artifact support for Shellable (including an initial
attempt at a BuildSpec helper as requested in awslabs/aws-cdk#2039).

Add multiple input artifact support for the GitHub publisher. The
tag and release notes will be taken from the primary artifact, the other
artifacts will be attached to the same release.

This is in preparation of publishing downloadable reference docs to
GitHub.

### CODE REVIEW

File: lib/build-spec.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  }
+
+  public render(): BuildSpecStruct {
+    return this.spec;
+  }
+}
+
+export interface MergeOptions {
+  /**
+   * Rename the primary artifact on the right-hand-side of the merge operation, if present
```
Comment (eladb) [inline lib/build-spec.ts]: Mention that this is required if the rhs config includes a primary artifact

File: lib/shellable.ts [hunk 2]
```diff
@@ -145,23 +146,6 @@ export interface AssumeRole {
   externalId?: string;
 }
 
-/**
- * Properties used to create a Shellable
- */
-export interface ShellableProps extends ShellableOptions {
```
Comment (eladb) [inline lib/shellable.ts]: What’s up here? No props?

Comment (rix0rrr) [reply lib/shellable.ts]: Duplicate definition.

Comment (eladb) [reply lib/shellable.ts]: Not sure what that means? What am I missing?

Comment (rix0rrr) [reply lib/shellable.ts]: We had 2 definitions of this interface in the same file. They were both exactly the same.

Comment (eladb) [reply lib/shellable.ts]: 👍

File: lib/build-spec.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  }
+}
+
+export interface SimpleBuildSpecProps {
+  preBuild?: string[];
+  build?: string[];
+
+  /**
+   * Where the directories for each artifact are
+   *
```
Comment (eladb) [inline lib/build-spec.ts]: I kind of liked the previous API better that optimized for a single primary artifact:

```ts
interface SimpleBuildSpecProps {
  // ...
  artifactsDir: string;

  additionalArtifactDirs? : { [ ... ] }
}

**Pull request merged**